libs/ezsat/ezcommand.cc

```diff
@@ -0,0 +1,86 @@
+
+#include "ezcommand.h"
+
+#include "../../kernel/yosys.h"
+
+ezSATCommand::ezSATCommand(const std::string &cmd) : command(cmd) {}
+
+ezSATCommand::~ezSATCommand() {}
+
+bool ezSATCommand::solver(const std::vector<int> &modelExpressions, std::vector<bool> &modelValues, const std::vector<int> &assumptions)
+{
+#if !defined(YOSYS_DISABLE_SPAWN)
+	const std::string tempdir_name = Yosys::make_temp_dir(Yosys::get_base_tmpdir() + "/yosys-sat-XXXXXX");
+	const std::string cnf_filename = Yosys::stringf("%s/problem.cnf", tempdir_name.c_str());
+	const std::string sat_command = Yosys::stringf("%s %s", command.c_str(), cnf_filename.c_str());
+	FILE *dimacs = fopen(cnf_filename.c_str(), "w");
+
+	std::vector<int> modelIdx;
+	for (auto id : modelExpressions)
+		modelIdx.push_back(bind(id));
+	std::vector<std::vector<int>> extraClauses;
+	for (auto id : assumptions)
+		extraClauses.push_back({bind(id)});
+
+	printDIMACS(dimacs, false, extraClauses);
```

Comment:
Would there be issues with appending the extra clauses to the file from here, instead of introducing a new argument to `printDIMACS`? I see `printDIMACS` finds out the maximum term or clause number to arrange for uniform padding, but is that a requirement of the format or just a convenience for inspection?

Reply:
I preferred to extend printDIMACS rather than duplicate the writing code. You'd prefer not to change the API?

Reply:
It depends on how much code would be duplicated in practice. How much lines would you estimate that to be?

Reply:
Not that many actually, less than 10

Reply:
As pointed out by Jix, the number of clauses is in the header, so we cannot actually add this afterwards